### PR TITLE
Handle invalid Cargo manifest decoding errors

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- catch `tomllib.TOMLDecodeError` in the manifest reader CLI so invalid TOML fails fast
- add Python unit tests that exercise the script's field extraction, CLI options, and error paths

## Testing
- python -m unittest discover -s tests_python
- make check-fmt
- make lint
- make test

------
https://chatgpt.com/codex/tasks/task_e_68de5d76681c83229e66aaeefe15c1a8

## Summary by Sourcery

Improve the manifest reader CLI to handle invalid TOML, refactor release workflows to use a new Python upload-release-assets action, update lint settings, and add comprehensive tests for both tools

New Features:
- Catch invalid TOML decode errors in the manifest reader CLI to fail fast
- Add a Python-based upload_release_assets script and composite GitHub Action for uploading release artefacts

Enhancements:
- Replace inline Bash upload logic in release workflows with the reusable upload-release-assets action
- Add detailed docstrings to read_manifest CLI functions

CI:
- Update release and release-dry-run workflows to emit metadata outputs and invoke the upload-release-assets action
- Add task-tags configuration to ruff.toml for linting

Tests:
- Add unit tests for read_manifest covering field extraction, CLI options, and error paths
- Add tests for upload_release_assets script including asset discovery, collision and empty-file handling, and dry-run mode

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[tests_python/test_read_manifest.py](https://github.com/leynos/netsuke/pull/179/files/79d68cacbc5bf99704e8e71c08e6b31c440fe3fb#diff-ba8ca0324778dfba9278998aad8e998ae0efc207b1a18689f2ac3bd90714ac88)

Comment on lines +99 to +109

~~~python
    def test_main_reports_missing_manifest(self) -> None:
        missing = self.temp_path / "missing.toml"
        result = subprocess.run(
            [sys.executable, str(SCRIPT_PATH), "name", "--manifest-path", str(missing)],
            check=False,
            capture_output=True,
            text=True,
        )
        self.assertNotEqual(result.returncode, 0)
        self.assertIn("does not exist", result.stderr)
        self.assertEqual(result.stdout, "")
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/69281/delta?repo-id=1160466&review-id=179&biomarker=Code+Duplication&filename=tests_python%2Ftest_read_manifest.py&method=)
The module contains 2 functions with similar structure: ReadManifestTests.test_main_reports_invalid_toml,ReadManifestTests.test_main_reports_missing_manifest

File: tests_python/test_read_manifest.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        missing = self.temp_path / "missing.toml"
+        result = subprocess.run(
+            [sys.executable, str(SCRIPT_PATH), "name", "--manifest-path", str(missing)],
+            check=False,
+            capture_output=True,
+            text=True,
+        )
+        self.assertNotEqual(result.returncode, 0)
+        self.assertIn("does not exist", result.stderr)
+        self.assertEqual(result.stdout, "")
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests_python/test_read_manifest.py]: Has this now been resolved?

File: tests_python/test_read_manifest.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    def test_get_field_returns_version(self) -> None:
+        manifest = {"package": {"name": "netsuke", "version": "1.2.3"}}
+        self.assertEqual(self.module.get_field(manifest, "version"), "1.2.3")
+
+    def test_get_field_raises_when_missing(self) -> None:
+        manifest = {"package": {"name": "netsuke"}}
+        with self.assertRaises(KeyError):
+            self.module.get_field(manifest, "version")
+
+    def test_main_reads_manifest_path_argument(self) -> None:
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests_python/test_read_manifest.py]: Has this now been resolved?

**Comment by leynos** [comment]
@sourcery-ai review

File: scripts/upload_release_assets.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    size = path.stat().st_size
+    if size <= 0:
+        message = f"Artefact {path} is empty"
+        raise AssetError(message)
+    return size
+
+
+def _register_asset(asset_name: str, path: Path, seen: dict[str, Path]) -> None:
+    previous = seen.get(asset_name)
+    if previous:
```
[the remark this answers is not in the document]
Comment (leynos) [reply scripts/upload_release_assets.py]: Has this now been resolved?

**Comment by leynos** [comment]
@sourcery-ai review
